src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java

```diff
@@ -399,6 +403,23 @@ public Object handle(
         }
     }
 
+    public static class WayPointBlockPlacementHandler implements IPlacementHandler
+    {
+        @Override
+        public Object handle(
+                @NotNull final World world, @NotNull final BlockPos pos, @NotNull final IBlockState blockState,
+                @Nullable final AbstractEntityAIStructure<?> placer)
+        {
+            if (!(blockState.getBlock() instanceof BlockWaypoint))
+            {
+                return ActionProcessingResult.IGNORE;
+            }
+
+            placer.addWayPoint(pos);
+            return blockState;
+        }
+    }
+
     public static class ChestPlacementHandler implements IPlacementHandler
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 36 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)